shell/src/main/java/alluxio/cli/fs/command/PinCommand.java

```diff
@@ -49,8 +50,13 @@ protected void runPlainPath(AlluxioURI path, CommandLine cl)
       throws AlluxioException, IOException {
     String[] args = cl.getArgs();
     // args[0] is the path, args[1] to args[end] is the list of possible media to pin
-    List<String> mediumTypes = Arrays.asList(Arrays.copyOfRange(args, 1, args.length));
-    FileSystemCommandUtils.setPinned(mFileSystem, path, true, mediumTypes);
+    List<String> pinnedMediumTypes = Arrays.asList(Arrays.copyOfRange(args, 1, args.length));
+    List<String> availableMediumList = mFsContext.getPathConf(path).getList(
+        PropertyKey.MASTER_TIERED_STORE_GLOBAL_MEDIUMTYPES, ",");
+    if (!availableMediumList.containsAll(pinnedMediumTypes)) {
+      throw new IllegalArgumentException("Invalid medium to pin the file");
```

Comment:
Could you print which one is invalid / give a list of valid ones?

Reply:
done.